pkg/resource/resid.go

```diff
@@ -28,18 +28,36 @@ type ResId struct {
 	gvk schema.GroupVersionKind
 	// original name of the resource before transformation.
 	name string
+	// namePrefix of the resource
+	prefix string
 }
 
 // NewResId creates new resource identifier
-func NewResId(g schema.GroupVersionKind, n string) ResId {
-	return ResId{gvk: g, name: n}
+func NewResId(g schema.GroupVersionKind, n ...string) ResId {
+	if len(n) >= 2 {
+		return ResId{gvk: g, name: n[0], prefix: n[1]}
+	}
+	return ResId{gvk: g, name: n[0]}
 }
 
+// String of ResId based on GVK, name and prefix
 func (n ResId) String() string {
+	var fields []string
+	for _, s := range []string{n.gvk.Group, n.gvk.Version, n.gvk.Kind, n.prefix, n.name} {
+		if s != "" {
```

Comment:
It's helpful (for debugging, logging, etc) if the String() method is relatively unique.  I.e. the String() methods of two different instances should not be equal.  

The risk in leaving out the empty strings is you can get two different objects with the same String() result.  In practice supposedly the only string that could be empty is the prefix - but stuff that shouldn't happen often does.  So for String(), its better to just join the fields as is, empty or no, to avoid any chance of this.  So yes, you might get two underscores next to each other, but that's a clear signal as to what is missing.

If you want this behavior for some other purpose, just give it a name other than String(), e.g. PrettyString or something.